game-core/src/main/java/games/strategy/engine/data/unit/ability/CombatUnitAbility.java

```diff
@@ -0,0 +1,106 @@
+package games.strategy.engine.data.unit.ability;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+
+/** Unit Ability for combat situations */
+@Data
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Builder(toBuilder = true)
+public class CombatUnitAbility {
+
+  enum DiceType {
+    // Use AA dice (attackAA, offensiveAttackAA)
+    AA,
```

Comment:
Context seems to be missing here by what is meant for 'dice'. Why is the 'dice' comment significant, and what is meant by 'AA dice'?

Reply:
I'm currently not envisioning the unit abilities to specify the dice properties (strength and rolls).  That information would stay on the unit itself.  I'm also not envisioning changing the existing names of the dice properties (`attack`, `defense`, `attackAA`, `offensiveAtackAA`, `bombard`, `airAttack`, `airDefense`).  That's why I have this `DiceType` value class with only three initial items.

But, if we want to make changes to the dice properties, then this class will need to be reworked.  I don't think the dice properties should be on the `CombatUnitAbility`.  That would make it hard to share `CombatUnitAbility` between units.

I also thought about having this `DiceType` be an aspect of the `BattlePhase` instead of the `CombatUnitAbility`.  The existing battle phases and combat unit abilities act like that (all combat unit abilities in the AA phase use the AA dice type).

Reply:
Thank you for the context @trevan . I think I will struggle for some time with the term 'DiceType' as cognitively it does not represent at all in TripleA what I would think it should mean. To be clear, dice type simply refers to the number of sides, or does it also refer to the number of rolls?

Reply:
side-comment, if we can come up with better terminology, it would likely help, I think there is a terminology and model mismatch that makes this confusing.

Reply:
> To be clear, dice type simply refers to the number of sides, or does it also refer to the number of rolls?

It refers to the strength and the rolls.

In the current engine, when you have AA guns, the dice properties are `attackAA`, `attackAAmaxDieSides`, `offensiveAttackAA`, `offensiveAttackAAmaxDieSides`, and `maxAAattacks`.  When units bombard, the dice properties are `bombard` or `attack` if `bombard` is missing.  Outside of AA and bombard, units use `attack`, `defense`, `attackRolls`, and `defenseRolls`.

I needed a way to indicate which of those sets of properties should be used in the `CombatUnitAbility`.  So I created 3 groups and called them `AA`, `BOMBARD`, and `NORMAL` and described them as `DiceType`.

Reply:
Perhaps it could be called 'StrengthAndRolls'? We have that value object and it then seemingly is a more direct name. 

Dice type for me refers either to the number of dice sides or is seemingly a different concept altogether more related to perhaps phases, or even things like technology roll vs a battle roll.

Reply:
It is more similar to the `CombatValue` value objects.  There are `AaDefenseCombatValue`, `AaOffenseCombatValue`, `BombardmentCombatValue`, `MainDefenseCombatValue`, `MainOffenseCombatValue`.  `DiceType` is used to decide which of the CombatValue classes should be used (AA, Bombardment, or Main).  So maybe call this `CombatValueType`?

Reply:
`CombatValueType` or `CombatValue` both seem like improvements to me FWIW

Reply:
I've renamed it to `CombatValueType` and added a link to `CombatValue` in the javadoc - https://github.com/triplea-game/triplea/pull/8646/commits/75fb84b36a20a193f5e58c70e2d4adc2f37af77d